retrievers/merger_retriever.go

```diff
@@ -0,0 +1,44 @@
+package retrievers
+
+import (
+	"context"
+
+	"github.com/tmc/langchaingo/callbacks"
+	"github.com/tmc/langchaingo/schema"
+)
+
+var _ schema.Retriever = &MergerRetriever{}
+
+// MergerRetriever is a retriever that merges the results of multiple retrievers.
+type MergerRetriever struct {
+	Retrievers       []schema.Retriever
+	CallbacksHandler callbacks.Handler
+}
+
+// NewMergerRetriever creates a new MergerRetriever.
+func NewMergerRetriever(
+	retrievers []schema.Retriever,
+) MergerRetriever {
+	return MergerRetriever{
+		Retrievers:       retrievers,
+		CallbacksHandler: nil,
+	}
+}
+
+func (m *MergerRetriever) GetRelevantDocuments(ctx context.Context, query string) ([]schema.Document, error) {
```

Comment:
could write a comment about it, input and output?

Reply:
please add this, then LGTM.

Reply:
> please add this, then LGTM.

Update.